### PR TITLE
scaffold horizontal bars

### PR DESCRIPTION
**Pull request by elassari19**
Screen Shot 2024-12-23 at 23 52 43
Screen Shot 2024-12-23 at 23 52 46

### CODE REVIEW

File: src/components/Bars/components/HorizontalBars.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import React from "react";
+
+type Props = React.SVGAttributes<SVGSVGElement> & {
+	children?: React.ReactNode;
+};
+
+export const HorizontalBars = ({ children, className }: Props) => {
+	const context = useGraph();
+
+	if (!context) return null;
```
Comment (ShanonJackson) [inline src/components/Bars/components/HorizontalBars.tsx:15]: No `if(!context)` check required anymore. Context is always there or it throws now under new system.

File: src/components/XAxis/XAxis.tsx [hunk 2]
```diff
@@ -13,20 +14,28 @@ type Props = {
 	ticks?: { from?: From; to?: To; jumps?: Jumps };
 	title?: ReactNode;
 	description?: ReactNode;
+	horizontal?: boolean;
 };
 
-export const XAxis = ({ title, description }: Props) => {
+export const XAxis = ({ title, description, horizontal }: Props) => {
```
Comment (ShanonJackson) [inline src/components/XAxis/XAxis.tsx:20]: No horizontal prop on this please.
Should be sufficient for someone who wants horizontal bars to:

1. Swap the x/y of their dataset around i.e {x: "January" becomes {y: "January" if you want horizontal.
2. use `<Bars horizontal={true}/>`


This removes the need for changes to XAxis/YAxis in ways that only make sense for <Bars/> but not <Lines/>

File: src/components/YAxis/YAxis.tsx [hunk 3]
```diff
@@ -13,9 +13,10 @@ type Props = {
 	ticks?: { from?: From; to?: To; jumps?: Jumps };
 	title?: ReactNode;
 	description?: ReactNode;
+	horizontal?: boolean;
 };
 
-export const YAxis = ({ title, description }: Props) => {
+export const YAxis = ({ title, description, horizontal }: Props) => {
```
Comment (ShanonJackson) [inline src/components/YAxis/YAxis.tsx:19]: No horizontal prop on this please.
Should be sufficient for someone who wants horizontal bars to:

1. Swap the x/y of their dataset around i.e {x: "January" becomes {y: "January" if you want horizontal.
2. use `<Bars horizontal={true}/>`


This removes the need for changes to XAxis/YAxis in ways that only make sense for <Bars/> but not <Lines/>


**Same as above just reposting here**

File: src/components/Bars/Bars.tsx [hunk 4]
```diff
@@ -1,7 +1,11 @@
 import React from "react";
 import { VerticalBars } from "@/components/Bars/components/VerticalBars";
-import { ComponentProps } from "react";
+import { HorizontalBars } from "./components/HorizontalBars";
 
-export const Bars = (props: ComponentProps<typeof VerticalBars>) => {
-	return <VerticalBars {...props} />;
+type ComponentProps = {
+	horizontal?: boolean;
```
Comment (ShanonJackson) [inline src/components/Bars/Bars.tsx:6]: This is exactly what i have in mind.

File: src/app/bar-graph/page.tsx [hunk 5]
```diff
@@ -23,31 +24,34 @@ export default function Page() {
...
-			<div className={"border-[1px] h-full border-dotted border-white"}>
+			<div className={"h-full border-dotted border border-black dark:border-white"}>
 				<Graph data={[{ name: "My First Dataset", data: MOCK_DATA }]} gap={{ top: 15, left: 15, right: 30, bottom: 15 }}>
-					<YAxis ticks={{ from: 0 }} />
+					<YAxis horizontal={bars} ticks={{ from: 0 }} />
 					<GridLines {...gridline} />
-					<Bars />
-					<XAxis />
+					<Bars horizontal={bars} />
+					<XAxis horizontal={bars} />
```
Comment (ShanonJackson) [inline src/app/bar-graph/page.tsx:51]: See comments below about XAxis/YAxis taking `horizontal` thank you.

File: src/components/Bars/components/HorizontalBars.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		<svg
+			viewBox={`0 0 ${context.viewbox.x} ${context.viewbox.y}`}
+			className={cx("[grid-area:graph] h-full w-full", className)}
+			preserveAspectRatio={"none"}
+		>
+			{bars.data.map((bar, index) => {
+				return (
+					<path
+						key={index}
+						d={`M 0 ${bar.x - barHeight / 2} l${bar.y} 0 l0 ${barHeight} l${-bar.y} 0 Z`}
```
Comment (ShanonJackson) [inline src/components/Bars/components/HorizontalBars.tsx:47]: Small but please don't draw a line where `border` goes for GridLines.
So for horizontal should be:

Line outwards right.
Line down.
Line inwards left.

Hope this makes sense if not iv'e attached a screenshot.
image

File: src/components/Bars/components/HorizontalBars.tsx (hunk 1 above)
Comment (elassari19) [reply src/components/Bars/components/HorizontalBars.tsx]: done

File: src/components/XAxis/XAxis.tsx (hunk 2 above)
Comment (elassari19) [reply src/components/XAxis/XAxis.tsx]: done

File: src/components/YAxis/YAxis.tsx (hunk 3 above)
Comment (elassari19) [reply src/components/YAxis/YAxis.tsx]: done

File: src/components/Bars/components/HorizontalBars.tsx (hunk 6 above)
Comment (elassari19) [reply src/components/Bars/components/HorizontalBars.tsx]: done

**Comment by elassari19** [comment]
I made the requirement changes, please check them and let me know if there is more

**Pull request merged**